pydarn/io/borealis_formats.py

```diff
@@ -0,0 +1,283 @@
+# Copyright 2018 SuperDARN
+# Authors: Marina Schmidt
+"""
+This file contains several class with the fields that pertain to
+SuperDARN Borealis HDF5 files.
+
+Classes:
+--------
+BorealisRawacf
+BorealisFitacf
+BorealisBfiq
+BorealisOutputPtrsIq
+BorealisRawrf
+
+Notes:
+------
+Borealis stage iq data will not be included here (intermediate writes
+can be done between rawrf and output_ptrs_iq processing)
+
+"""
+
+import numpy as np
+
+
+class BorealisRawacf():
+    """
+    Class containing Borealis Rawacf fields
+    """
+    single_element_types = {
+        # Identifies the version of Borealis that made this data.
+        "borealis_git_hash": np.unicode_,
+        # Timestamp of when the record was written. Seconds since epoch.
+        "timestamp_of_write": np.float64,
+        "experiment_id": np.int64,  # Number used to identify experiment.
+        "experiment_name": np.unicode_,  # Name of the experiment file.
+        "experiment_comment": np.unicode_,  # Comment about the whole experiment
+        # Additional text comment that describes the slice.
+        "slice_comment": np.unicode_,
+        # Number of slices in the experiment at this integration time.
+        "num_slices": np.int64,
+        "station": np.unicode_,  # Three letter radar identifier.
+        # Number of sampling periods in the integration time.
+        "num_sequences": np.int64,
+        # range gate separation (equivalent distance between samples)
+        "range_sep": np.float32,
+        # Round trip time of flight to first range in microseconds.
+        "first_range_rtt": np.float32,
+        "first_range": np.float32,  # Distance to first range in km.
+        # Sampling rate of the samples being written to file in Hz.
+        "rx_sample_rate": np.float64,
+        # Designates if the record is the first in a scan.
+        "scan_start_marker": np.bool_,
+        "int_time": np.float32,  # Integration time in seconds.
+        "tx_pulse_len": np.uint32,  # Length of the pulse in microseconds.
+        # The minimum spacing between pulses, spacing between pulses is always a
+        "tau_spacing": np.uint32,
+        # multiple of this in microseconds.
+        "main_antenna_count": np.uint32,  # Number of main array antennas.
+        # Number of interferometer array antennas.
+        "intf_antenna_count": np.uint32,
+        # The frequency used for this experiment slice in kHz.
+        "freq": np.uint32,
+        # C data type of the samples such as complex float.
+        "samples_data_type": np.unicode_,
+        # data normalization factor determined by the filter scaling in the decimation scheme.
+        "data_normalization_factor": np.float64
+    }
+
+    array_dtypes = {
+        "pulses": np.uint32,  # The pulse sequence in units of the tau_spacing.
```

Comment:
I'm not sure what docs you mean. It may not be applicable because we changed the names of fields for Borealis, to names we thought were more understandable.

Reply:
Yea.. that is something I need to write up on the DMAP SuperDARN formats for RST. 

You can reference the old docuhttps://superdarn.github.io/rst/superdarn/src.doc/rfc/0008.htmlmentation but it is not a big deal. 

I just saw the mention of tau and then "units".

Reply:
tau is not from the old formats though, it won't be mentioned in the RST doc. tau_spacing is another field and pulses is given as a multiple of tau_spacing. I can reword the comment to say multiples instead of units